Scriptorium-CSC309/web
# finished fetch all code templates feature

**Summary**

An visitor can now fetch all code templates filtered by code content, title, and/or tags.

Note that this currently doesn't sort by relevance (the matched amount); this could be a future feature to implement.
**Review Notes**
Thanks to @Ram-Raghav-S , I implemented this feature by modifying his original code for get-comments.

**Testing**

**Test authenticated fetch:**
1. Login via login API
2. Save code template via its `POST save-code-template`
3. Save another code template via its sub-request `POST save-code-template-1`
4. Fetch using `GET fetch-all-code-templates` (should see two, or more results if existing templates also use given tags)
5. Fetch filtered using `GET filter-code-templates` (should see one less result than step 4)

**Test un-authenticated fetch:**
after running test for authenticated, do
1. Fetch using `GET visitor-fetch-all` API
Should see at least the same results as step 5 of previous test (provided the auth'ed user has userId 1).

Issues to be closed by this PR: 
- https://github.com/Scriptorium-CSC309/web/issues/25
- https://github.com/Scriptorium-CSC309/web/issues/22
